### PR TITLE
Fixes #9524, adjusts compatibility for glibc packages and adds file lists for glibc_build232, and adds max_glibc property

### LINKED ISSUE #9524: Missing filelists and overmatching compatibility for glibc_build232 and glibc_build233
As per title, there are no filelists for `glibc_build232`. 

Additionally, it has compatibility marked as `all` despite not providing binaries for `i686`, and with our current glibc setup it will never provide binaries for `i686`, so we should probably mark it as incompatible.

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #9524 
- Also adds `max_glibc` property to packages.
- Update compatibility logic to also report reasons for incompatibility.
### Run the following to get this pull request's changes locally for testing.
```bash
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=glibc_compat_fixup crew update \
&& yes | crew upgrade ; crew list compatible | grep glibc_build223 ; crew install glibc_build223
```

### CODE REVIEW

**Comment by satmandu** [comment]
Clearly I have hit a workflow unit test edge condition here. We shouldn't try installing older glibc on newer containers. 😅

**Comment by satmandu** [comment]
Might have to add a max_glibc property... (I say in jest...)

**Comment by satmandu** [comment]
Will work on arm container issue later...

**Comment by satmandu** [comment]
Ah, also broke the `min_glibc` detection code in the unit test... Can we run `ruby` in the GitHub CI runner directly?

File: .github/workflows/Unit-Test.yml [hunk 1]
```diff
@@ -62,16 +62,16 @@ jobs:
...
             echo "PR #${PR_NUMBER} has these x86_64 compatible packages: ${X86_64_PACKAGES}" && \
             export ARMV7L_PACKAGES="$(for p in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|armv7l" packages/${p}.rb && echo ${p}; done | xargs)" && \
             echo "PR #${PR_NUMBER} has these ARMv7L compatible packages: ${ARMV7L_PACKAGES}" && \
             [[ $ARMV7L_PACKAGES ]] || exit 0 && \
             if [[ $GLIBC_237_COMPATIBLE_PACKAGES ]] ; then armv7l_CONTAINER="satmandu/crewbuild:strongbad-armv7l.m126"; else armv7l_CONTAINER="satmandu/crewbuild:fievel-armv7l.m91"; fi &&\
-            export container_cmdline="/usr/bin/sudo -u chronos LD_LIBRARY_PATH=/usr/lib:/lib /usr/local/bin/bash -c \"export ALL_CHANGED_FILES='${ALL_CHANGED_FILES}' && \
+                                                                                                                                export container_cmdline="/usr/bin/sudo -u chronos LD_LIBRARY_PATH=/usr/local/lib:/usr/lib:/lib /usr/local/bin/bash -c \"export ALL_CHANGED_FILES='${ALL_CHANGED_FILES}' && \
```
Comment (uberhacker) [inline .github/workflows/Unit-Test.yml:74]: Too much  indentation?

Comment (satmandu) [reply .github/workflows/Unit-Test.yml]: Definitely. This is what happens when you are editing over ash from a phone...😬

**Review by uberhacker — APPROVED** [review]
